### PR TITLE
Loop condition should be re-evaluated after continue in while and do-while statements

### PR DESCRIPTION
**Pull request by fedeazzato**
Changed how while and do-while lower continue statements.

In the following sample:

```
{
    var i = 0
    while i < 5
    {
        i = i + 1
        if i == 5
            continue
    }
    print(string(i))
}

```
was printing 6 as a result, while the equivalent C# would print 5.

### CODE REVIEW

File: src/Minsk/CodeAnalysis/Binding/Binder.cs [hunk 1]
```diff
@@ -239,15 +239,15 @@ private BoundStatement BindIfStatement(IfStatementSyntax syntax)
         private BoundStatement BindWhileStatement(WhileStatementSyntax syntax)
         {
             var condition = BindExpression(syntax.Condition, TypeSymbol.Bool);
-            var body = BindLoopBody(syntax.Body, out var breakLabel, out var continueLabel);
-            return new BoundWhileStatement(condition, body, breakLabel, continueLabel);
+            var body = BindLoopBody(syntax.Body, out var bodyLabel, out var breakLabel, out var continueLabel);
```
Comment (terrajobst) [inline src/Minsk/CodeAnalysis/Binding/Binder.cs]: I don't think we should add the `bodyLabel` to the bound nodes. The reason that we have `breakLabel` and `continueLabel` is that we can access them during binding. The `bodyLabel` should just be generated during lowering, like it already happens with labels like `checkLabel`.

File: .vscode/launch.json [hunk 2]
```diff
@@ -10,9 +10,9 @@
             "request": "launch",
             "preLaunchTask": "build",
             // If you have changed target frameworks, make sure to update the program path.
-            "program": "${workspaceFolder}/src/mc/bin/Debug/netcoreapp2.1/mc.dll",
+            "program": "${workspaceFolder}/src/mi/bin/Debug/netcoreapp2.1/mi.dll",
             "args": [],
-            "cwd": "${workspaceFolder}/src/mc",
+            "cwd": "${workspaceFolder}/src/mi",
```
Comment (terrajobst) [inline .vscode/launch.json]: Good catch! :-)

File: src/mi/Repl.cs [hunk 3]
```diff
@@ -214,6 +214,7 @@ private void HandleEscape(ObservableCollection<string> document, SubmissionView
             document.Clear();
             document.Add(string.Empty);
             view.CurrentLine = 0;
+            view.CurrentCharacter = 0;
```
Comment (terrajobst) [inline src/mi/Repl.cs]: Why is this needed?

Comment (fedeazzato) [reply src/mi/Repl.cs]: This proposed change is to fix an ArgumentOutOfRangeException. The setter of [CurrentLine](https://github.com/terrajobst/minsk/blob/7abf535b0d1dd825718584c713372b97196ed215/src/mi/Repl.cs#L107) checks if the assigned new line is different from the current value.
When handling escape for single line input (or for multi-line input but the cursor is already at the first line), the current line is already 0, and thus the character isn't moved to 0. That leaves the character in the same place as it was before handling escape. If at that point you try to use backspace to clear the white spaces, an exception is thrown, because the caret is in an invalid position.

File: src/Minsk/CodeAnalysis/Binding/Binder.cs (hunk 1 above)
Comment (fedeazzato) [reply src/Minsk/CodeAnalysis/Binding/Binder.cs]: Ok, I was trying to have more descriptive label names when seeing the lowered program. I reverted these changes.

**Comment by fedeazzato** [comment]
Hi Immo

Just wanted to say that I reverted the commit that bothered you instead of rebasing the pull request because I didn't want to hide my mistake. Seeing now the mess that it made to your master history, I feel silly for that. I'm sorry for the noise.

Thanks for this series. It has been a lot of fun following it. It doesn't bother me at all, but as I've seen you struggle to try to say my name, Fede is a short for Federico, a Spanish name, which [Google translate](https://translate.google.com/#view=home&op=translate&sl=es&tl=es&text=Fede) pronounces quite well. [Bing translator](https://www.bing.com/translator) pronunciation is perfect, but it seems to lack the option to have a permalink that fills the inputs.

Thanks again for the time and efforts you put into the series. I hadn't seen compiler code since college, and it has been a lot of fun refreshing some memories, and learning a lot of new things.

I can't wait to see how you're going to make a debugger for Minsk language that works well in the console :-)
